lib/routes/password.js

```diff
@@ -147,15 +150,47 @@ module.exports = function (
         var sessionTokenId = request.payload.sessionToken
         var password = new Password(authPW, authSalt, verifierVersion)
         var wantsKeys = requestHelper.wantsKeys(request)
-        var account, verifyHash, sessionToken, keyFetchToken
+        var account, verifyHash, sessionToken, keyFetchToken, verifiedStatus
 
-        return changePassword()
+        getSessionVerificationStatus()
+          .then(changePassword)
           .then(notifyAccount)
           .then(createSessionToken)
           .then(createKeyFetchToken)
           .then(createResponse)
           .done(reply, reply)
 
+        function getSessionVerificationStatus() {
+          if (sessionTokenId) {
+            return Tokens.SessionToken.fromHex(sessionTokenId)
```

Comment:
I didn't realize the client was passing in the raw sessionToken data blob here, rather than just the token id.  We generally try to avoid passing the raw sessionToken data over the wire because it's a plaintext credential....sorry, I should have noticed this in review of the original change-password PR.  I think we can push ahead with it for now but will file a follow-up issue to address this.